diff --git a/snapshotsna.pas b/snapshotsna.pas
index 14b7ef0..a680e67 100644
--- a/snapshotsna.pas
+++ b/snapshotsna.pas
@@ -136,7 +136,7 @@ begin
             N := Stream.Position;
             if (Stream.Seek(Int64(KB48), TSeekOrigin.soCurrent) = KB48 + N)
               and (Stream.Read(State.PC, 2) = 2)
-              and (Stream.Read(B, 1) = 1)
+              and (Stream.Read(B{%H-}, 1) = 1)
               and (Stream.Seek(-(Int64(KB48) + 3), TSeekOrigin.soCurrent) = N)
             then begin
               State.Set7ffd(B);
@@ -239,7 +239,6 @@ var
   MS16: TMemoryStream;
               
   W: Word;
-  WR: WordRec absolute W;
 
 begin
   Result := False;
diff --git a/snapshotz80.pas b/snapshotz80.pas
index 351d5a6..52d3ed9 100644
--- a/snapshotz80.pas
+++ b/snapshotz80.pas
@@ -325,7 +325,7 @@ begin
                   // read one more byte if LeH3 = 55
                   if Header2.LeH3 <> 55 then
                     HeadersRead := 3
-                  else if Stream.Read(B, 1) = 1 then
+                  else if Stream.Read(B{%H-}, 1) = 1 then
                     HeadersRead := 4;
                 end;
               end else
@@ -527,7 +527,7 @@ begin
       Header1.IFF2 := 1;
 
     Header1.ViB := (WordRec(State.IR).Lo shr 7)
-      or ((State.BorderColour and %111) shl 1)
+      or ((State.BorderColour and Byte(%111)) shl 1)
       or %100000
       ;
 
diff --git a/unitcolourpalette.pas b/unitcolourpalette.pas
index 52e741f..763b5a4 100644
--- a/unitcolourpalette.pas
+++ b/unitcolourpalette.pas
@@ -10,7 +10,7 @@ interface
 uses
   Classes, SysUtils, Types, UnitSpectrumColourMap, CommonFunctionsLCL,
   UnitCommonSpectrum, UnitFormForOptionsBasic, UnitOptions, Forms, Controls,
-  ExtCtrls, Graphics, StdCtrls, Dialogs, Buttons, ButtonPanel, LCLType, LCLIntf,
+  ExtCtrls, Graphics, StdCtrls, Dialogs, Buttons, LCLType, LCLIntf,
   LazUTF8;
 
 type
diff --git a/unitfilesna.pas b/unitfilesna.pas
index 59566de..086401e 100644
--- a/unitfilesna.pas
+++ b/unitfilesna.pas
@@ -8,8 +8,7 @@ unit UnitFileSna;
 interface
 
 uses
-  Classes, SysUtils, UnitSpectrum, UnitMemory, Z80Processor, FastIntegers,
-  SoundChipAY_3_8912;
+  Classes, SysUtils, UnitSpectrum, UnitMemory, Z80Processor, SoundChipAY_3_8912;
 
 const
   KB48 = 48 * 1024;
@@ -226,13 +225,12 @@ class function TSnapshotInternalSwan.GetScreenMem(const Stream: TStream;
   var ScreenMemArr: array of Byte): Boolean;
 var
   State: TSpectrumInternalState;
-  N: Integer;
 begin
   if Assigned(Stream) and (Length(ScreenMemArr) = 6912)
     and (Stream.Size >= SizeOf(TSpectrumInternalState) + Length(ScreenMemArr))
   then begin
     Stream.Position := 0;
-    if Stream.Read(State, SizeOf(State)) = SizeOf(State) then begin
+    if Stream.Read(State{%H-}, SizeOf(State)) = SizeOf(State) then begin
       if State.ShadowScreenDisplay then
         Stream.Seek(7 * Int64(KB16), TSeekOrigin.soCurrent);
 
@@ -250,7 +248,6 @@ var
   WasPaused: Boolean;
   RamBanksCount: Integer;
   Mem: TMemory;
-  I: Integer;
 
 begin
   Result := False;
diff --git a/unitformhistorysnapshots.lfm b/unitformhistorysnapshots.lfm
index 7582e73..f1c595a 100644
--- a/unitformhistorysnapshots.lfm
+++ b/unitformhistorysnapshots.lfm
@@ -9,7 +9,7 @@ object FormHistorySnapshots: TFormHistorySnapshots
   OnCreate = FormCreate
   OnDestroy = FormDestroy
   OnShow = FormShow
-  LCLVersion = '2.2.7.0'
+  LCLVersion = '3.0.0.2'
   object Panel15: TPanel
     AnchorSideLeft.Control = Owner
     AnchorSideTop.Control = Owner
@@ -139,10 +139,10 @@ object FormHistorySnapshots: TFormHistorySnapshots
             AnchorSideTop.Control = Panel12
             AnchorSideRight.Control = Panel12
             AnchorSideRight.Side = asrBottom
-            Left = 435
+            Left = 437
             Height = 19
             Top = 0
-            Width = 156
+            Width = 154
             Anchors = [akTop, akRight]
             Caption = 'Restore selected snapshot'
             TabOrder = 0
diff --git a/unitformhistorysnapshots.pas b/unitformhistorysnapshots.pas
index 3d8a8ea..9a4782e 100644
--- a/unitformhistorysnapshots.pas
+++ b/unitformhistorysnapshots.pas
@@ -9,9 +9,9 @@ interface
 
 uses
   Classes, SysUtils, Types, UnitFileSna, UnitHistorySnapshots,
-  CommonFunctionsLCL, UnitKeyMaps, UnitFormPressAKey, UnitControlScreenBitmap,
+  CommonFunctionsLCL, UnitControlScreenBitmap,
   UnitSpectrumColoursBGRA, UnitFrameHistorySnapshotOptions, Forms, Controls,
-  Graphics, Dialogs, ExtCtrls, StdCtrls, Spin, LMessages, ButtonPanel;
+  Graphics, Dialogs, ExtCtrls, StdCtrls, LMessages, ButtonPanel;
 
 type
   TFormHistorySnapshots = class(TForm)
diff --git a/unitformpressakey.pas b/unitformpressakey.pas
index a5119f6..4db7ef4 100644
--- a/unitformpressakey.pas
+++ b/unitformpressakey.pas
@@ -17,7 +17,7 @@ type
     Label1: TLabel;
     procedure FormCreate(Sender: TObject);
     procedure FormDestroy(Sender: TObject);
-    procedure FormKeyDown(Sender: TObject; var Key: Word; Shift: TShiftState);
+    procedure FormKeyDown(Sender: TObject; var Key: Word; {%H-}Shift: TShiftState);
     procedure FormShow(Sender: TObject);
   private
     FKey: Word;
diff --git a/unitframekeymappings.pas b/unitframekeymappings.pas
index e27bebf..3cb7865 100644
--- a/unitframekeymappings.pas
+++ b/unitframekeymappings.pas
@@ -10,7 +10,7 @@ interface
 uses
   Classes, SysUtils, UnitFrameOneKeyMap, UnitKeyMapRecords, UnitFormPressAKey,
   CommonFunctionsLCL, UnitFormForOptionsBasic, UnitOptions, Forms, Controls,
-  Graphics, Dialogs, ExtCtrls, ButtonPanel, StdCtrls, LazUTF8;
+  Graphics, Dialogs, ExtCtrls, StdCtrls, LazUTF8;
 
 type
   TFrameKeyMappings = class(TFrame)
@@ -56,7 +56,7 @@ type
     procedure SaveSchemeOnClick(Sender: TObject);
     procedure RemoveSchemeOnClick(Sender: TObject);
     procedure ResetToDefaultOnClick(Sender: TObject);
-    procedure Close(Sender: TObject; var CloseAction: TCloseAction);
+    procedure Close(Sender: TObject; var {%H-}CloseAction: TCloseAction);
 
   public
     constructor Create(TheOwner: TComponent); override;
diff --git a/unitframeonekeymap.pas b/unitframeonekeymap.pas
index 7ec89e5..fd38aec 100644
--- a/unitframeonekeymap.pas
+++ b/unitframeonekeymap.pas
@@ -278,7 +278,6 @@ procedure TControlKeyMappings.RemovePCKey(AValue: Word);
 var
   N: Integer;
   F: TFrameOnePCKeyMapping;
-  I: Integer;
 begin
   N := FrameMappingsMap.IndexOf(AValue);
   if N >= 0 then begin
@@ -319,7 +318,7 @@ var
   SKL: TFrameOnePCKeyMapping.TSpectrumKeyControls;
 begin
   N := (FrameMappingsMap.Count * 7) div 5 + 2;
-  SetLength(KMRecs, N);
+  SetLength(KMRecs{%H-}, N);
   K := 0;
   for I := 0 to FrameMappingsMap.Count - 1 do begin
     SKL := FrameMappingsMap.Data[I].SpectrumKeys;
@@ -554,7 +553,7 @@ begin
       WE := 0;
       for I := 0 to High(SpectrumKeys) do begin
         OSC := SpectrumKeys[I];
-        OSC.GetPreferredSize(W, H);
+        OSC.GetPreferredSize(W{%H-}, H{%H-});
         WE := WE + W;
         if Assigned(OSC1) then begin
           WE := WE + SpcW;
diff --git a/unitframespectrummodel.pas b/unitframespectrummodel.pas
index 0d761c8..bbee57a 100644
--- a/unitframespectrummodel.pas
+++ b/unitframespectrummodel.pas
@@ -9,7 +9,7 @@ interface
 
 uses
   Classes, SysUtils, UnitSpectrum, UnitOptions, Forms, Controls, ExtCtrls,
-  StdCtrls, Graphics;
+  Graphics;
 
 type
 
diff --git a/unitinputlibrarypathdialog.pas b/unitinputlibrarypathdialog.pas
index d62671e..8aba6f6 100644
--- a/unitinputlibrarypathdialog.pas
+++ b/unitinputlibrarypathdialog.pas
@@ -34,7 +34,7 @@ type
     procedure MakeExtensionsFilter;
     function DoCheckLoad: Boolean;
     procedure FormOnCloseQuery(Sender: TObject; var CanClose: Boolean);
-    procedure FormOnClose(Sender: TObject; var CloseAction: TCloseAction);
+    procedure FormOnClose(Sender: TObject; var {%H-}CloseAction: TCloseAction);
   public
     constructor Create(TheOwner: TComponent); override;
     constructor CreateLibraryPathDialog(AOwner: TComponent; S: String;
diff --git a/unitrecentfiles.pas b/unitrecentfiles.pas
index 579e6b6..b81e1d9 100644
--- a/unitrecentfiles.pas
+++ b/unitrecentfiles.pas
@@ -126,7 +126,7 @@ var
 begin
   Result := False;
   K := Length(FRecentFiles);
-  SetLength(Arr, K);
+  SetLength(Arr{%H-}, K);
   J := 0;
   for I := Low(FRecentFiles) to High(FRecentFiles) do begin
     if not SameFileName(FRecentFiles[I], S) then begin
diff --git a/unitszx.pas b/unitszx.pas
index 35f772d..678939d 100644
--- a/unitszx.pas
+++ b/unitszx.pas
@@ -295,7 +295,7 @@ var
 begin
   Result := False;
   if (BlockSize = SizeOf(AyRec))
-     and (Stream.Read(AyRec, SizeOf(AyRec)) = SizeOf(AyRec))
+     and (Stream.Read(AyRec{%H-}, SizeOf(AyRec)) = SizeOf(AyRec))
   then begin
     for I := 0 to 15 do begin
       Szx.State.AyState.Registers[I] := AyRec.AyRegisters[I];
@@ -1238,7 +1238,6 @@ var
 
 var
   BlockRP: TZxstRamPage;
-  BlockTape: TZxstTape;
   I, J: Integer;
   TapePlayer: TTapePlayer;
 
diff --git a/unittapeplayer.pas b/unittapeplayer.pas
index 5a8c1be..d630750 100644
--- a/unittapeplayer.pas
+++ b/unittapeplayer.pas
@@ -77,7 +77,7 @@ type
     class function GetTapeType: TTapeType; virtual; abstract;
 
     class procedure RegisterTapePlayerClass(TapePlayerClass: TTapePlayerClass); static;
-    class function GetNextBlockClass(const Stream: TStream): TTapeBlockClass; virtual;
+    class function GetNextBlockClass(const {%H-}Stream: TStream): TTapeBlockClass; virtual;
     class function CheckIsMyClass(const Stream: TStream): Boolean; virtual; abstract;
 
   private
@@ -462,10 +462,7 @@ begin
 end;
 
 class function TTapePlayer.GetDefaultExtension: String;
-var
-  TapeType: TTapeType;
 begin
-  TapeType := GetTapeType;
   WriteStr(Result, GetTapeType);
   Delete(Result, 1, 2);
   Result := LowerCase(Result);
